codegenerator/cli/src/hbs_templating/codegen_templates.rs

```diff
@@ -349,55 +337,120 @@ impl EntityRecordTypeTemplate {
 pub struct EventTemplate {
     pub name: CapitalizedOptions,
     pub params: Vec<EventParamTypeTemplate>,
-    pub indexed_params: Vec<EventParamTypeTemplate>,
-    pub body_params: Vec<EventParamTypeTemplate>,
     pub topic0: String,
+    pub decode_hyper_fuel_data_code: String,
+    pub convert_hyper_sync_event_args_code: String,
+    pub data_type: String,
+    pub data_schema_code: String,
 }
 
 impl EventTemplate {
-    pub fn from_config_event(config_event: &system_config::Event) -> Result<Self> {
-        let name = config_event.name.to_capitalized_options();
-        let params = config_event
-            .get_event_inputs()
-            .iter()
-            .map(|input| {
-                let res_type = abi_to_rescript_type(&input.into());
-                let js_name = input.name.to_string();
-                EventParamTypeTemplate {
-                    res_name: make_res_name(&js_name),
-                    js_name,
-                    default_value_rescript: res_type.get_default_value_rescript(),
-                    default_value_non_rescript: res_type.get_default_value_non_rescript(),
-                    res_type: res_type.to_string(),
-                    res_schema_code: res_type.to_rescript_schema(),
-                    is_eth_address: res_type == RescriptTypeIdent::Address,
-                    is_indexed: input.indexed,
-                    type_rescript_skar_decoded_param: res_type.to_string_decoded_skar(),
-                }
-            })
-            .collect::<Vec<_>>();
+    const DECODE_HYPER_FUEL_DATA_CODE: &'static str =
+        "(_) => Js.Exn.raiseError(\"HyperFuel decoder not implemented\")";
 
+    pub fn generate_convert_hyper_sync_event_args_code(params: &Vec<EventParam>) -> String {
```

Comment:
Can we handle the empty vec in this function so the context makes a bit more sense?

So check if it's empty and return the `(Utils.magic: HyperSyncClient.Decoder.decodedEvent => eventArgs)` here if so.